packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1111,15 +1119,16 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
 
   /// Called when this object is removed from the tree.
   ///
-  /// The framework calls this method whenever it removes this [State] object
-  /// from the tree. In some cases, the framework will reinsert the [State]
-  /// object into another part of the tree (e.g., if the subtree containing this
-  /// [State] object is grafted from one location in the tree to another). If
-  /// that happens, the framework will ensure that it calls [build] to give the
-  /// [State] object a chance to adapt to its new location in the tree. If
-  /// the framework does reinsert this subtree, it will do so before the end of
-  /// the animation frame in which the subtree was removed from the tree. For
-  /// this reason, [State] objects can defer releasing most resources until the
+  /// If the [widget] or one of its ancestors has a [GlobalKey], the framework
+  /// calls this method whenever it removes this [State] object from the tree.
```

Comment:
actually it always calls it when removing the widget, not just when it has a global key.

Reply:
You're right, I got confused.